pkg/operator/resource.go

```diff
@@ -0,0 +1,56 @@
+package operator
+
+import (
+	"context"
+	"fmt"
+
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/serializer"
+	"k8s.io/client-go/dynamic"
+	"sigs.k8s.io/yaml"
+
+	monitoringv1 "github.com/prometheus-operator/prometheus-operator/pkg/apis/monitoring/v1"
+
+	"github.com/openshift/library-go/pkg/operator/events"
+	"github.com/openshift/library-go/pkg/operator/resource/resourceapply"
+	"github.com/openshift/library-go/pkg/operator/resource/resourceread"
+)
+
+var (
+	monitoringScheme = runtime.NewScheme()
+	monitoringCodecs = serializer.NewCodecFactory(monitoringScheme)
+)
+
+func init() {
+	if err := monitoringv1.AddToScheme(monitoringScheme); err != nil {
+		panic(err)
+	}
+}
+
+func ReadServiceMonitorV1OrDie(objBytes []byte) *monitoringv1.ServiceMonitor {
+	requiredObj, err := runtime.Decode(monitoringCodecs.UniversalDecoder(monitoringv1.SchemeGroupVersion), objBytes)
+	if err != nil {
+		panic(err)
+	}
+	return requiredObj.(*monitoringv1.ServiceMonitor)
+}
+
+// ApplyServiceMonitor applies the ServiceMonitor.
+func ApplyServiceMonitor(ctx context.Context, client dynamic.Interface, recorder events.Recorder, required *monitoringv1.ServiceMonitor, cache resourceapply.ResourceCache) (*unstructured.Unstructured, bool, error) {
```

Comment:
nit: Probably we agree that this is not ideal and should be located in library-go. But we don't do any complicated operations on this resource and maybe we can use just as unstructured. But this is not a blocker for me.

Reply:
It seemed like a lot of operations on unstructured, so having a type is more transparent. And yeah it might be good to have something like this in library-go.